File: LeetCodeSolutions/Java/2401_LongestNiceSubarray.java
class Solution {
    public int longestNiceSubarray(int[] nums) {
        int len = nums.length, mask = 0, ans = 0, left = 0, cur = 0;
        for(int right = 0; right<len; right++){
            while((mask & nums[right]) != 0){
                mask^=nums[left];
                left++;
            }
            mask |= nums[right];
            cur = right-left+1;
            if(ans < cur){
                ans = cur;
            }
        }
        return ans;
    }
}

File: LeetCodeSolutions/C/2401_LongestNiceSubarray.c
int longestNiceSubarray(int* nums, int numsSize) {
    int ans = 0, cur =0, left = 0, mask = 0;
    for(int right = 0; right < numsSize; right ++){
        while((mask & nums[right]) != 0){
            mask^=nums[left];
            left++;
        }
        mask|=nums[right];
        int cur = right - left+1;
        ans = cur>ans?cur:ans;
    }
    return ans;
}
